source/common/access_log/access_log_formatter.cc

```diff
@@ -308,11 +308,16 @@ std::string RequestHeaderFormatter::format(const Http::HeaderMap& request_header
   return HeaderFormatter::format(request_headers);
 }
 
-StartTimeFormatter::StartTimeFormatter(const std::string& format) : format_(format) {}
+StartTimeFormatter::StartTimeFormatter(const std::string& format)
+    : date_formatter_(std::make_unique<Envoy::DateFormatter>(format)) {}
 
 std::string StartTimeFormatter::format(const Http::HeaderMap&, const Http::HeaderMap&,
                                        const RequestInfo::RequestInfo& request_info) const {
-  return AccessLogDateTimeFormatter::fromTimeWithFormat(request_info.startTime(), format_);
+  if (date_formatter_->formatString().empty()) {
```

Comment:
Instead of having a conditional here, just construct date_formatter_ with the default format string if the passed-in one is empty.

Reply:
The only issue with a default date format string is that the default `START_TIME` currently does a little string formatting on top of the date in `AccessLogDateTimeFormatter::fromTime` to get milliseconds, which you can't get from [date format strings](http://en.cppreference.com/w/cpp/chrono/c/strftime): 
```
  static DateFormatter default_date_format("%Y-%m-%dT%H:%M:%S");

  return fmt::format(
      "{}.{:03d}Z", default_date_format.fromTime(time),
      std::chrono::duration_cast<std::chrono::milliseconds>(time.time_since_epoch()).count() %
          1000);
```
and I thought it best not to duplicate this block in the `StartTimeFormatter`. So in any case, there'd have to be branching somewhere to determine whether to apply the millisecond string formatting (or drop milliseconds from the default datetime). Thoughts?